### PR TITLE
Add reCaptcha enterprise support

### PR DESCRIPTION
**Pull request by AwesomeNipun**
### Proposed changes in this pull request

This PR is for adding the support for reCaptcha enterprise.
For the reCaptcha flows, in addition to the reCaptcha v2 and v3 support, reCaptcha enterprise support is added.

- If reCaptcha enterprise is enabled, the verification method is different from v2 and v3. This is mainly handled in the CaptchaUtil.
- Additional variables needed for reCaptcha enterprise support are added.

### When should this PR be merged

Can be merged once reviewed and approved.

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/internal/CaptchaDataHolder.java [hunk 1]
```diff
@@ -45,6 +47,10 @@ public class CaptchaDataHolder {
 
     private String reCaptchaSecretKey;
 
+    private String reCaptchaAPIKey;
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/internal/CaptchaDataHolder.java:50]: Reuse the existing secret key for the ReCaptcha Enterprise

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/resources/api.identity.recovery.yaml [hunk 2]
```diff
@@ -790,6 +790,8 @@ definitions:
     properties:
       reCaptchaEnabled:
         type: boolean
+      reCaptchaEnterpriseEnabled:
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/resources/api.identity.recovery.yaml:793]: Add a UI config for ReCaptcha Enterprise enabled and use it for get-captcha API

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/internal/CaptchaDataHolder.java [hunk 3]
```diff
@@ -37,6 +37,8 @@ public class CaptchaDataHolder {
 
     private boolean reCaptchaEnabled;
 
+    private boolean reCaptchaEnterpriseEnabled;
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/internal/CaptchaDataHolder.java:40]: reCaptchaEnabled and reCaptchaEnterpriseEnabled are two different options. So, the name should imply that.

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 4]
```diff
@@ -269,60 +271,132 @@ public static Map<String, String> getClaimValues(User user, int tenantId,
...
         CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build();
-        HttpPost httppost = new HttpPost(CaptchaDataHolder.getInstance().getReCaptchaVerifyUrl());
         final double scoreThreshold = CaptchaDataHolder.getInstance().getReCaptchaScoreThreshold();
+        boolean isRecaptchaEnterpriseEnabled = CaptchaDataHolder.getInstance().isReCaptchaEnterpriseEnabled();
 
-        List<BasicNameValuePair> params = Arrays.asList(new BasicNameValuePair("secret", CaptchaDataHolder
-                .getInstance().getReCaptchaSecretKey()), new BasicNameValuePair("response", reCaptchaResponse));
-        httppost.setEntity(new UrlEncodedFormEntity(params, StandardCharsets.UTF_8));
+        HttpPost httppost;
+        if (!isRecaptchaEnterpriseEnabled) {  // for recaptcha v2 and v3
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:278]: Change this to isRecaptchaEnterpriseEnabled

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 5]
```diff
@@ -269,60 +271,132 @@ public static Map<String, String> getClaimValues(User user, int tenantId,
...
-                        log.debug("reCAPTCHA v3 response { timestamp:" +
-                                verificationResponse.get("challenge_ts") + ", action: " +
-                                verificationResponse.get("action") + ", score: " + score + " }");
+            } catch (IOException e) {
+                throw new CaptchaServerException("Unable to read the verification response.", e);
+            } catch (ClassCastException e) {
+                throw new CaptchaServerException("Unable to cast the response value.", e);
+            }
+
+        } else {   // for recaptcha enterprise
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:334]: modify comment. Capital letter and fullstop.

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 6]
```diff
@@ -269,60 +271,132 @@ public static Map<String, String> getClaimValues(User user, int tenantId,
...
+
+            String recaptchaUrl = CaptchaDataHolder.getInstance().getReCaptchaVerifyUrl();
+            String projectID = CaptchaDataHolder.getInstance().getReCaptchaProjectID();
+            String APIKey = CaptchaDataHolder.getInstance().getReCaptchaAPIKey();
+            String siteKey = CaptchaDataHolder.getInstance().getReCaptchaSiteKey();
+
+            String verifyUrl = recaptchaUrl + "/v1/projects/" + projectID + "/assessments?key=" + APIKey;
+            httppost = new HttpPost(verifyUrl);
+
+            httppost.setHeader("Content-Type", "application/json");
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:344]: add constants

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 7]
```diff
@@ -269,60 +271,132 @@ public static Map<String, String> getClaimValues(User user, int tenantId,
...
+            try {
+                try (InputStream in = responseEntity.getContent()) {
+                    JsonObject verificationResponse = new JsonParser().parse(IOUtils.toString(in)).getAsJsonObject();
+                    if (verificationResponse == null) {
+                        throw new CaptchaClientException("Error receiving reCaptcha response from the server");
                     }
-                    if (score < scoreThreshold) {
-                        throw new CaptchaClientException("reCaptcha score is less than the threshold.");
+                    JsonObject tokenProperties = verificationResponse.get("tokenProperties").getAsJsonObject();
+                    boolean success = tokenProperties.get("valid").getAsBoolean();
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:373]: use constant for valid

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 8]
```diff
@@ -269,60 +271,132 @@ public static Map<String, String> getClaimValues(User user, int tenantId,
...
+                    // Whether this request was a valid reCAPTCHA token.
+                    if (!success) {
+                        throw new CaptchaClientException("reCaptcha token is invalid. Error:" +
+                                verificationResponse.get("error-codes"));
                     }
-                } else {
-                    if (log.isDebugEnabled()) {
-                        log.debug("reCAPTCHA v2 response { timestamp:" +
-                                verificationResponse.get("challenge_ts") + " }");
+                    if (riskAnalysis.get("score") != null) {
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:382]: Shall we use a constant for `score`?

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java [hunk 9]
```diff
@@ -483,19 +499,45 @@ public static HttpResponse makeCaptchaVerificationHttpRequest(ReCaptchaResponseT
...
-                new BasicNameValuePair("response", reCaptchaResponse.getToken()));
-        httppost.setEntity(new UrlEncodedFormEntity(params, StandardCharsets.UTF_8));
+        HttpPost httppost;
 
-        try {
-            response = httpclient.execute(httppost);
-        } catch (IOException e) {
-            RecoveryUtil.handleBadRequest(String.format("Unable to get the verification response : %s", e.getMessage()),
-                    Constants.STATUS_INTERNAL_SERVER_ERROR_MESSAGE_DEFAULT);
+        if (!reCaptchaEnterpriseEnabled){   // for Recaptcha V2 and V3
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java:508]:
```suggestion
        if (!reCaptchaEnterpriseEnabled) {   
```
Fix formattings. Shall we reverse the condition to get rid of the negation? and also shall we add the comment in a new line? Fix the formattings of comments as well.

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java [hunk 10]
```diff
@@ -483,19 +499,45 @@ public static HttpResponse makeCaptchaVerificationHttpRequest(ReCaptchaResponseT
...
-        String reCaptchaSecretKey = properties.getProperty(CaptchaConstants.RE_CAPTCHA_SECRET_KEY);
         String reCaptchaVerifyUrl = properties.getProperty(CaptchaConstants.RE_CAPTCHA_VERIFY_URL);
+        boolean reCaptchaEnterpriseEnabled =
+                Boolean.valueOf(properties.getProperty(CaptchaConstants.RE_CAPTCHA_ENTERPRISE_ENABLED));
         CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build();
-        HttpPost httppost = new HttpPost(reCaptchaVerifyUrl);
-        List<BasicNameValuePair> params = Arrays.asList(new BasicNameValuePair("secret", reCaptchaSecretKey),
-                new BasicNameValuePair("response", reCaptchaResponse.getToken()));
-        httppost.setEntity(new UrlEncodedFormEntity(params, StandardCharsets.UTF_8));
+        HttpPost httppost;
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java:506]:
```suggestion
        HttpPost httpPost;
```

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java [hunk 11]
```diff
@@ -483,19 +499,45 @@ public static HttpResponse makeCaptchaVerificationHttpRequest(ReCaptchaResponseT
...
+            httppost.setEntity(new UrlEncodedFormEntity(params, StandardCharsets.UTF_8));
+
+            try {
+                response = httpclient.execute(httppost);
+            } catch (IOException e) {
+                RecoveryUtil.handleBadRequest(String.format("Unable to get the verification response : %s",
+                        e.getMessage()), Constants.STATUS_INTERNAL_SERVER_ERROR_MESSAGE_DEFAULT);
+            }
+
+        } else{ // for Recaptcha Enterprise
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java:523]:
```suggestion
        } else { 
```

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/CaptchaApiServiceImpl.java [hunk 12]
```diff
@@ -89,20 +95,42 @@ public Response verifyCaptcha(ReCaptchaResponseTokenDTO reCaptchaResponse, Strin
...
         ReCaptchaVerificationResponseDTO reCaptchaVerificationResponseDTO = new ReCaptchaVerificationResponseDTO();
 
-        try {
+        if (!reCaptchaEnterpriseEnabled) {
+
+            try {
+                if (entity == null) {
+                    RecoveryUtil.handleBadRequest("ReCaptcha verification response is not received.",
+                            Constants.STATUS_INTERNAL_SERVER_ERROR_MESSAGE_DEFAULT);
+                } else {
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/CaptchaApiServiceImpl.java:104]: We can remove the else here and move the code to the left right?

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 13]
```diff
@@ -269,60 +271,132 @@ public static Map<String, String> getClaimValues(User user, int tenantId,
...
+            }
+
+            try {
+                try (InputStream in = entity.getContent()) {
+                    JsonObject verificationResponse = new JsonParser().parse(IOUtils.toString(in)).getAsJsonObject();
+                    if (verificationResponse == null) {
+                        throw new CaptchaClientException("Error receiving reCaptcha response from the server");
+                    }
+                    boolean success = verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS) != null
+                            && verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS).getAsBoolean();
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:304]:
```suggestion
                    boolean success = Boolean.parseBoolean(CaptchaConstants.CAPTCHA_SUCCESS);
```
WDYT?

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 14]
```diff
@@ -446,6 +520,32 @@ private static UserStoreManager getUserStoreManagerForUser(String userName,
 
     private static void setReCaptchaConfigs(Properties properties) {
 
+        boolean recaptchaEnterpriseEnabled =
+                Boolean.parseBoolean(properties.getProperty(CaptchaConstants.RE_CAPTCHA_ENTERPRISE_ENABLED));
+
+        if (recaptchaEnterpriseEnabled){    // reCaptcha Enterprise require API key and Project ID
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:526]: formatting issues. Check other places too.

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java (hunk 6 above)
Comment (AwesomeNipun) [reply components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:344]: Constants not available in the CaptchaConstants.

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 15]
```diff
@@ -269,62 +283,134 @@ public static Map<String, String> getClaimValues(User user, int tenantId,
...
+            String siteKey = CaptchaDataHolder.getInstance().getReCaptchaSiteKey();
+            String secretKey = CaptchaDataHolder.getInstance().getReCaptchaSecretKey();
 
-        HttpResponse response;
-        try {
-            response = httpclient.execute(httppost);
-        } catch (IOException e) {
-            throw new CaptchaServerException("Unable to get the verification response.", e);
-        }
+            String verifyUrl = recaptchaUrl + "/v1/projects/" + projectID + "/assessments?key=" + secretKey;
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:297]: try to move to a private method

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java [hunk 16]
```diff
@@ -456,6 +558,20 @@ private static UserStoreManager getUserStoreManagerForUser(String userName,
 
     private static void setReCaptchaConfigs(Properties properties) {
 
+        String reCaptchaType = properties.getProperty(CaptchaConstants.RE_CAPTCHA_TYPE);
+        if (!StringUtils.isBlank(reCaptchaType)) {
+            CaptchaDataHolder.getInstance().setReCaptchaType(reCaptchaType);
+        }
+
+        if (reCaptchaType.equals(CaptchaConstants.RE_CAPTCHA_TYPE_ENTERPRISE)) {
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java:566]: use `CONSTANT_STRING.equals(variableString)` not the other way around

File: components/org.wso2.carbon.identity.api.user.recovery/src/gen/java/org/wso2/carbon/identity/recovery/endpoint/dto/ReCaptchaPropertiesDTO.java [hunk 17]
```diff
@@ -57,14 +72,15 @@ public void setReCaptchaAPI(String reCaptchaAPI) {
...
+
 
   @Override
   public String toString()  {
     StringBuilder sb = new StringBuilder();
     sb.append("class ReCaptchaPropertiesDTO {\n");
-    
+
     sb.append("  reCaptchaEnabled: ").append(reCaptchaEnabled).append("\n");
+    sb.append("  reCaptchaEnterpriseEnabled: ").append(reCaptchaType).append("\n");
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.user.recovery/src/gen/java/org/wso2/carbon/identity/recovery/endpoint/dto/ReCaptchaPropertiesDTO.java:83]: Is the "reCaptchaEnterpriseEnabled" correct here?

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java [hunk 18]
```diff
@@ -469,6 +470,11 @@ private static Properties validateCaptchaConfigs(Properties properties) {
             RecoveryUtil.handleBadRequest(String.format("%s is not found ", CaptchaConstants.RE_CAPTCHA_VERIFY_URL),
                     Constants.STATUS_INTERNAL_SERVER_ERROR_MESSAGE_DEFAULT);
         }
+        if (!StringUtils.isBlank(reCaptchaType) && reCaptchaType.equals(CaptchaConstants.RE_CAPTCHA_TYPE_ENTERPRISE) &&
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java:473]:
```suggestion
        if (CaptchaConstants.RE_CAPTCHA_TYPE_ENTERPRISE.equals(recaptchaType) &&
```

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/CaptchaApiServiceImpl.java [hunk 19]
```diff
@@ -64,6 +65,9 @@ public Response getCaptcha(String captchaType, String recoveryType, String tenan
             reCaptchaPropertiesDTO.setReCaptchaEnabled(true);
             reCaptchaPropertiesDTO.setReCaptchaKey(properties.getProperty(CaptchaConstants.RE_CAPTCHA_SITE_KEY));
             reCaptchaPropertiesDTO.setReCaptchaAPI(properties.getProperty(CaptchaConstants.RE_CAPTCHA_API_URL));
+            if (!StringUtils.isBlank(reCaptchaType)) {
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/CaptchaApiServiceImpl.java:68]: Do we need to check for isBlank here?

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/CaptchaApiServiceImpl.java [hunk 20]
```diff
@@ -89,20 +94,43 @@ public Response verifyCaptcha(ReCaptchaResponseTokenDTO reCaptchaResponse, Strin
         HttpEntity entity = response.getEntity();
         ReCaptchaVerificationResponseDTO reCaptchaVerificationResponseDTO = new ReCaptchaVerificationResponseDTO();
 
-        try {
+        if (StringUtils.isBlank(reCaptchaType) || reCaptchaType.equals(CaptchaConstants.
+                RE_CAPTCHA_TYPE_DEFAULT)){
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/CaptchaApiServiceImpl.java:98]: use `CONSTANT_STRING.equals(variableString)` not the other way around

File: components/org.wso2.carbon.identity.api.user.recovery/src/test/resources/repository/conf/identity/captcha-config.properties [hunk 21]
```diff
@@ -21,6 +21,9 @@
 # Enable Google reCAPTCHA
 recaptcha.enabled=true
 
+# Google reCAPTCHA type
+recaptcha.type=default
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.user.recovery/src/test/resources/repository/conf/identity/captcha-config.properties:25]: IMO we don't need a default value

File: components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java [hunk 22]
```diff
@@ -469,6 +470,11 @@ private static Properties validateCaptchaConfigs(Properties properties) {
             RecoveryUtil.handleBadRequest(String.format("%s is not found ", CaptchaConstants.RE_CAPTCHA_VERIFY_URL),
                     Constants.STATUS_INTERNAL_SERVER_ERROR_MESSAGE_DEFAULT);
         }
+        if (CaptchaConstants.RE_CAPTCHA_TYPE_ENTERPRISE.equals(reCaptchaType) &&
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/Utils/RecoveryUtil.java:473]: Shall we add a comment here explaining whats happening here?

File: components/org.wso2.carbon.identity.captcha/src/test/resources/testng.xml [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
+
+<suite name="User-Endpoint-Test-Suite">
+    <test name="Util-Tests" preserve-order="true" parallel="false">
+        <classes>
+            <class name="org.wso2.carbon.identity.captcha.util.CaptchaUtilTest"/>
+        </classes>
+    </test>
+</suite>
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.captcha/src/test/resources/testng.xml:26]: newline

**Comment by Lakshan-Banneheke** [comment]
This PR is continued in https://github.com/wso2-extensions/identity-governance/pull/756

**Pull request closed without merging**